deepchecks/nlp/text_data.py

```diff
@@ -222,56 +257,51 @@ def n_samples(self) -> int:
     def metadata(self) -> pd.DataFrame:
         """Return the metadata of for the dataset."""
         if self._metadata is None:
-            raise DeepchecksValueError('Metadata does not exist, please set it first using the set_metadata method')
+            raise ValueError(
+                'TextData does not contain metadata, add it by using '
+                '"set_metadata" function'
+            )
         return self._metadata
 
     @property
-    def metadata_types(self) -> t.Dict[str, str]:
-        """Return the metadata types of for the dataset."""
-        if self._metadata_types is None:
-            raise DeepchecksValueError('Metadata does not exist, please set it first using the set_metadata method')
-        return self._metadata_types
-
-    def set_metadata(self, metadata: pd.DataFrame, metadata_types: t.Optional[t.Dict[str, str]] = None):
-        """Set metadata for the dataset.
-
-        Parameters
-        ----------
-        metadata : pd.DataFrame
-            Metadata of the provided textual samples.
-        metadata_types : t.Optional[t.Dict[str, str]] , default : None
-            The types of the metadata columns. Can be either 'numeric' or 'categorical'.
-            If not provided, will be inferred automatically.
-        """
+    def cat_metadata(self) -> t.List[str]:
```

Comment:
change to full name - metadata_categorical_column_names

Reply:
changed to `categorical_metadata_columns`